### PR TITLE
feat(inheritance):  Add possible inheritance for  #24

### PR DESCRIPTION
**Pull request by xrobert35**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+language: node_js
```
Comment (xrobert35) [inline .travis.yml]: If you don't know it @olivierlsc  : You should try this  https://travis-ci.org

File: src/lib/swagger-express-ts/swagger.builder.ts [hunk 2]
```diff
@@ -195,7 +196,14 @@ export function build(buildDefinition: ISwaggerBuildDefinition): void {
...
   if (buildDefinition.models) {
-    SwaggerService.getInstance().setDefinitions(buildDefinition.models);
+    const models: any = {};
+    _.forOwn(buildDefinition.models, (value, key) => {
+      models[key] = {
+        name: key,
+        definition: value
+      };
+    });
+    SwaggerService.getInstance().setDefinitions(models);
```
Comment (xrobert35) [inline src/lib/swagger-express-ts/swagger.builder.ts]: I don't know if possible use  setDefinitions from "external"  but i've change a little bit the structure, so it make a breaking change

File: .travis.yml (hunk 1 above)
Comment (xrobert35) [reply .travis.yml]: actually the linter is not passing : https://travis-ci.org/xrobert35/swagger-express-ts

**Comment by jjaferson** [comment]
For some reason this merge is not introduced in the master branch anymore. it seems that there was a restructure of the folders and and change on the file  _src/lib/swagger-express-ts/api-model.decorator.ts got lost.


**After merging** 

export function ApiModel(args?: IApiModelArgs): ClassDecorator {
  return (target: any) => {
    // Extracting possible super class from prototype
    const protoType = Object.getPrototypeOf(target).name;
    let superClass = null;
    if (protoType !== "") {
      superClass = protoType;
    }

    SwaggerService.getInstance().addApiModel(args, target, superClass);
  };
}

**Current code**

export function ApiModel(args?: IApiModelArgs): ClassDecorator {
    return (target: any) => {
        SwaggerService.getInstance().addApiModel(args, target);
    };
}

**Comment by guilherme90** [comment]
@xrobert35  can you check this one? Me and others probatly needs that funcionallty

**Pull request merged**